cmd/plugins.go

```diff
@@ -0,0 +1,501 @@
+}
+
+// PubDep contains one entry of the pubspec.lock yaml list
+type PubDep struct {
+	Dependency  string
+	Description interface{}
+	Source      string
+	Version     string
+
+	// Fields set by hover
+	name    string
+	android bool
+	ios     bool
+	desktop bool
+	// optional description values
+	path string // correspond to the path field in lock file
+	host string // correspond to the host field in lock file
+	// contain a import.go.tmpl file used for import
+	autoImport bool
+	// the path/URL to the go code of the plugin is stored
+	pluginGoSource string
+	// whether or not the go plugin source code is located on another VCS repo.
+	standaloneImpl bool
+}
+
+func (p PubDep) imported() bool {
+	pluginImportOutPath := filepath.Join(buildPath, "cmd", fmt.Sprintf("import-%s-plugin.go", p.name))
+	if _, err := os.Stat(pluginImportOutPath); err == nil {
+		return true
+	}
+	return false
+}
+
+func (p PubDep) platforms() []string {
+	var platforms []string
+	if p.android {
+		platforms = append(platforms, "android")
+	}
+	if p.ios {
+		platforms = append(platforms, "ios")
+	}
+	if p.desktop {
+		platforms = append(platforms, buildPath)
+	}
+	return platforms
+}
+
+var pluginListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List golang platform plugins in the application",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) > 0 {
+			return errors.New("does not take arguments")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		assertInFlutterProject()
+		dependencyList, err := listPlatformPlugin()
+		if err != nil {
+			log.Errorf("%v", err)
+			os.Exit(1)
+		}
+
+		var hasNewPlugin bool
+		var hasPlugins bool
+		for _, dep := range dependencyList {
+			if !(dep.desktop || listAllPluginDependencies) {
+				continue
+			}
+
+			if hasPlugins {
+				fmt.Println("")
+			}
+			hasPlugins = true
+
+			fmt.Printf("     - %s\n", dep.name)
+			fmt.Printf("         version:   %s\n", dep.Version)
+			fmt.Printf("         platforms: [%s]\n", strings.Join(dep.platforms(), ", "))
+			if dep.desktop {
+				if dep.standaloneImpl {
+					fmt.Printf("         source:    This go plugin isn't maintained by the official plugin creator.\n")
+				}
+				if dep.imported() {
+					fmt.Println("         import:    [OK] The plugin is already imported in the project.")
+					continue
+				}
+				if dep.autoImport || dep.standaloneImpl {
+					hasNewPlugin = true
+					fmt.Println("         import:    [Missing] The plugin can be imported by hover.")
+				} else {
+					fmt.Println("         import:    [Manual import] The plugin is missing the import.go.tmpl file required for hover import.")
+				}
+				if dep.path != "" {
+					fmt.Printf("         dev:       Plugin replaced in go.mod to path: '%s'\n", dep.path)
+				}
+			}
+		}
+		if hasNewPlugin {
+			log.Infof(fmt.Sprintf("run `%s` to import the missing plugins!", log.Au().Magenta("hover plugins get").String()))
+		}
+	},
+}
+
+var pluginTidyCmd = &cobra.Command{
```

Comment:
@GeertJohan 
`clean` is often used as subcommand to remove everything that was created by the parent command. In this case, a user may expect `hover plugin clean` to remove everything that `hover plugin get` added. This may be confusing. I like how Go went with `tidy` (`go mod tidy`). It implies some stuff may be removed, but only when it was not needed anymore. Maybe we can go with `hover plugin tidy` ?

Reply:
Done